V2_MIGRATION_GUIDE.md

```diff
@@ -20,38 +19,42 @@ Version 2 only supports the [Authorization Code with Proof Key for Code Exchange
+The deprecated ability to authenticate using a `WebView` component has been removed. External browser applications will always be used instead for increased security and user trust. Please refer to [this blog post](https://auth0.com/blog/google-blocks-oauth-requests-from-embedded-browsers/) for additional information.
+
+## Removal of legacy Authentication API support
+
+Methods and classes specific to calling any [Authentication APIs](https://auth0.com/docs/api/authentication) categorized as Legacy have been removed in v2. The detailed listing of the removals are documented below.
 
 ## Request interfaces changes
 
-The `com.auth0.android.request` package defines the top-level interfaces for building and executing HTTP requests. Historically, a common issue has been the inability to add custom headers and request parameters to all request types. We've refactored the request interfaces to enable any request to be customized with additional headers and request parameters. Note that these parameters now need to be of type String.
+The `com.auth0.android.request` package defines the top-level interfaces for building and executing HTTP requests. Historically, a common issue has been the inability to add custom headers and request parameters to all request types. We've refactored the request interfaces to enable any request to be customized with additional headers and request parameters. Note that these parameters now need to be of type `String`.
 
-The top-level `com.auth0.android.request.Request` interface now specifies the following methods:
+The top-level `Request` interface now specifies the following methods:
 
-- `Request<T, U> addParameters(@NonNull Map<String, String> parameters)`
-- `Request<T, U> addParameter(@NonNull String name, @NonNull String value)`
-- `Request<T, U> addHeader(@NonNull String name, @NonNull String value)`
+- `public fun addParameters(parameters: Map<String, String>): Request<T, U>`
+- `public fun addParameter(name: String, value: String): Request<T, U>`
+- `public fun addHeader(name: String, value: String): Request<T, U>`
 
 As a result, the following interfaces have been removed:
 
-- `com.auth0.android.request.ParameterizableRequest`
-- `com.auth0.android.request.AuthRequest`
+- `ParameterizableRequest`
+- `AuthRequest`
 
-The `com.auth0.android.authentication.AuthenticationAPIClient` contains many changes to the return type of methods as a result. The full changes can be found below, but in summary:
+`AuthenticationAPIClient` contains many changes to the return type of methods as a result. The full changes can be found below, but in summary:
 
 - Any method that returned a `ParameterizableRequest` now returns a `Request`.
 - Any method that returned a `TokenRequest` now returns a `Request`.
 - Any method that returned a `AuthRequest` now returns an `AuthenticationRequest`.
 
```

Comment:
We should add one for the `DatabaseConnectionRequest`, which got removed in favor of `Request`. Maybe we can have the 3 that return `Request` in a single line https://github.com/auth0/Auth0.Android/pull/417/

Reply:
Updated to include `DatabaseConnectionRequest`, and consolidated to one line.